File: src/main/java/org/cimsbioko/navconfig/NavigatorModule.java
package org.cimsbioko.navconfig;

import org.cimsbioko.fragment.navigate.detail.DetailFragment;
import org.cimsbioko.navconfig.forms.Binding;
import org.cimsbioko.navconfig.forms.Launcher;

import java.util.List;
import java.util.Map;

public interface NavigatorModule {

    String getName();

    String getActivityTitle();

    String getLaunchLabel();

    String getLaunchDescription();

    Map<String, Binding> getBindings();

    List<Launcher> getLaunchers(String level);

    DetailFragment getDetailFragment(String level);

}

File: src/main/kotlin/org/cimsbioko/navconfig/NavigatorModule.kt
package org.cimsbioko.navconfig

interface NavigatorModule {
    val name: String
    val activityTitle: String
    val launchLabel: String
    val launchDescription: String
    val bindings: Map<String, Binding>
    fun getLaunchers(level: String): List<Launcher>
    fun getItemFormatter(level: String): ItemFormatter?
    fun getHierFormatter(level: String): HierFormatter?
}